### PR TITLE
Return a promise from registerAttributionSource method

### LINKED ISSUE #148: Improved error reporting for registerAttributionSource method
https://github.com/WICG/conversion-measurement-api/blob/main/event_attribution_reporting.md#registering-attribution-sources-with-javascript currently defines the `bool registerAttributionSource(AttributionSourceParams params)` method, which it says returns `false` if the parameters are invalid.

It would probably be more useful to throw a specific exception instead of returning a `bool`.

We could also consider changing this to a promise-based API, to cover the scenario in which the browser doesn't know whether a source will be logged when the API is invoked, e.g. because of storage limits.

### PR DESCRIPTION
**Pull request by apasel422**
Fixes #148.

### CODE REVIEW

File: event_attribution_reporting.md [hunk 1]
```diff
@@ -35,7 +35,7 @@ The browser will expose a new interface:
 
 ```
 [Exposed=Window] interface AttributionReporting {
-  bool registerAttributionSource(AttributionSourceParams params);
+  Promise<void> registerAttributionSource(AttributionSourceParams params);
```
Comment (apasel422) [inline event_attribution_reporting.md:38]: I'm not sure if this should be spelled `Promise<void>` or `Promise<undefined>`.

Comment (csharrison) [reply event_attribution_reporting.md:38]: I think void has been replaced by undefined in WebIDL from what I can see
https://heycam.github.io/webidl/#idl-undefined

Comment (apasel422) [reply event_attribution_reporting.md]: Done.

File: event_attribution_reporting.md [hunk 2]
```diff
@@ -50,7 +50,7 @@ dictionary AttributionSourceParams {
 }
 ```
 
-When invoked the browser will directly add the specified source to storage, and return whether the browser was successful in parsing the params.
+When invoked the browser will directly add the specified source to storage, and resolve or reject the promise based on whether the browser was successful in parsing the params and storing the source.
```
Comment (apasel422) [inline event_attribution_reporting.md:53]: We should think about whether failure to store the source leaks unintended information back to the caller.

Comment (csharrison) [reply event_attribution_reporting.md:53]: We should mention in the explainer that, due to privacy reasons, some failure states (those that depend on cross site information) may not show up as rejections.

Comment (apasel422) [reply event_attribution_reporting.md]: Done.

**Review by csharrison — APPROVED** [review]

**Pull request merged**